app/src/main/java/fr/free/nrw/commons/settings/SettingsFragment.java

```diff
@@ -85,7 +90,7 @@ public void onCreate(Bundle savedInstanceState) {
                 uploadLimit.setSummary(500 + "");
                 uploadLimit.setText(500 + "");
             } else {
-                editor.putInt(Prefs.UPLOADS_SHOWING, Integer.parseInt(newValue.toString()));
+                editor.putInt(Prefs.UPLOADS_SHOWING, value);
                 editor.putBoolean(Prefs.IS_CONTRIBUTION_COUNT_CHANGED,true);
                 uploadLimit.setSummary(newValue.toString());
             }
```

Comment:
While user sets upload limit to empty string, this patch set it to default value which is 100. However summary of uploadLimit field remains empty (users expectation is to see value 100 there). There should be an if-else condition to write 100 while newValue is non-number string.

Reply:
Doesn't the try-catch in line 74 take care of that case?

Reply:
It sets default value to 100, so that we can see 100 photos on contributions activity. However, it does not setSummary to 100, so after edit user had no idea that is sets to 100. After setting empty string, user sees empty summary as follow:
![device-2018-02-15-155107](https://user-images.githubusercontent.com/3127881/36257507-3c2145a8-1268-11e8-939d-9cc474ded4d6.png)